Upgraded app to AndroidX causes crash

**Describe the bug**
Upgraded project to AndroidX and camera no longer works as expected.

**To Reproduce**
Steps to reproduce the behavior:
1. Open up a screen that uses the camera

**Expected behavior**
Should work as expected and not crash.

**Using ProGuard**
- [ ] with ProGuard
- [x] without ProGuard

**Device (please complete the following information):**
 - Google Pixel 2
 - Android 8.1
 - 27

**Additional context / stacktrace**
`     Caused by: java.lang.IllegalStateException: Cannot invoke observe on a background thread
        at androidx.lifecycle.LiveData.assertMainThread(LiveData.java:443)
        at androidx.lifecycle.LiveData.observe(LiveData.java:171)
        at com.priyankvasa.android.cameraviewex.CameraConfigLiveData.observe$cameraViewEx_release(CameraConfigLiveData.kt:46)
        at com.priyankvasa.android.cameraviewex.Camera2.addObservers(Camera2.kt:490)
        at com.priyankvasa.android.cameraviewex.Camera2.<init>(Camera2.kt:82)
        at com.priyankvasa.android.cameraviewex.Camera2Api23.<init>(Camera2Api23.kt:32)
        at com.priyankvasa.android.cameraviewex.Camera2Api24.<init>(Camera2Api24.kt:31)
        at com.priyankvasa.android.cameraviewex.CameraView$camera$1.invokeSuspend(CameraView.kt:115)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:236)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:742))`

--- a/buildSrc/src/main/kotlin/Config.kt
+++ b/buildSrc/src/main/kotlin/Config.kt
@@ -20,7 +20,7 @@ object Config {
 
     private const val majorVersion = 3
     private const val minorVersion = 1
-    private const val patchVersion = 0
+    private const val patchVersion = 1
     private const val versionClassifier = ""
 
     val versionName: String

--- a/cameraViewEx/src/main/java/com/priyankvasa/android/cameraviewex/CameraView.kt
+++ b/cameraViewEx/src/main/java/com/priyankvasa/android/cameraviewex/CameraView.kt
@@ -101,11 +101,11 @@ class CameraView @JvmOverloads constructor(
             }
         )
 
-    private var camera: CameraInterface = runBlocking(coroutineContext) {
+    private var camera: CameraInterface = run {
 
         val cameraJob: Job = SupervisorJob(parentJob)
 
-        return@runBlocking when {
+        return@run when {
             Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP ->
                 Camera1(listenerManager, preview, config, cameraJob)
             Build.VERSION.SDK_INT < Build.VERSION_CODES.M ->
@@ -581,6 +581,13 @@ class CameraView @JvmOverloads constructor(
      * This is typically called from fragment's onDestroyView callback.
      */
     fun destroy() {
+        if (!isActive) {
+            listenerManager.onCameraError(
+                CameraViewException("CameraView instance already destroyed."),
+                ErrorLevel.Warning
+            )
+            return
+        }
         runBlocking(coroutineContext) {
             listenerManager.destroy()
             camera.destroy()

